src/main/java/bio/terra/service/auth/iam/IamResourceType.java

```diff
@@ -39,10 +47,12 @@ static IamResourceType fromValue(String text) {
   }
 
   public static IamResourceType fromEnum(IamResourceTypeEnum apiEnum) {
-    if (apiEnum == IamResourceTypeEnum.SPEND_PROFILE) {
-      return IamResourceType.SPEND_PROFILE;
+    for (IamResourceType b : IamResourceType.values()) {
+      if (b.getIamResourceTypeEnum().equals(apiEnum)) {
```

Comment:
Since enum members are all singletons it's OK (and IMO preferred) to compare them using `==`
```suggestion
      if (b.getIamResourceTypeEnum() == apiEnum) {
```